packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -120,8 +119,9 @@ export class RootDataObject extends DataObject<{ InitialState: RootDataObjectPro
     private async createDataObject<T extends IFluidLoadable>(dataObjectClass: DataObjectClass<T>): Promise<T> {
         const factory = dataObjectClass.factory;
         const packagePath = [...this.context.packagePath, factory.type];
-        const router = await this.context.containerRuntime.createDataStore(packagePath);
-        return requestFluidObject<T>(router, "/");
+        const dataStore = await this.context.containerRuntime.createDataStore(packagePath);
+        const entryPoint = await dataStore.entryPoint?.get();
+        return entryPoint as unknown as T;
```

Comment:
Do we need to fall back to the request pattern here if the entryPoint is undefined? Not familiar with how/where this RootDataObject is used. If it's used by FF consumers, seems like this could break them if they start using these bits but don't immediately start providing an entryPoint initialization function.

Reply:
Chatted with @ChumpChief and in theory, no consumer of FF should be using `RootDataObject` directly, it's not supposed to be part of the public API surface of the package. I created an ADO item to track its deprecation/removal-from-public-API, and will proceed with the change in this PR as it stands, since the factory that accompanies `RootDataObject` should already be providing the entrypoint initialization function (also part of this PR).